tiled/server/authentication.py

```diff
@@ -144,27 +145,41 @@ def create_refresh_token(session_id, secret_key, expires_delta):
     return encoded_jwt
 
 
-def decode_token(token, secret_keys):
+def decode_token(
+    token: str,
+    secret_keys: List[str],
+    proxied_authenticator: Optional[ProxiedOIDCAuthenticator] = None,
+):
     credentials_exception = HTTPException(
         status_code=HTTP_401_UNAUTHORIZED,
         detail="Could not validate credentials",
         headers={"WWW-Authenticate": "Bearer"},
     )
-    # The first key in settings.secret_keys is used for *encoding*.
-    # All keys are tried for *decoding* until one works or they all
-    # fail. They supports key rotation.
-    for secret_key in secret_keys:
-        try:
-            payload = jwt.decode(token, secret_key, algorithms=[ALGORITHM])
-            break
-        except ExpiredSignatureError:
-            # Do not let this be caught below with the other JWTError types.
-            raise
-        except JWTError:
-            # Try the next key in the key rotation.
-            continue
+    payload = ""
+    if proxied_authenticator:
+        payload = jwt.decode(
+            token,
+            key=proxied_authenticator.keys,
+            algorithms=proxied_authenticator.id_token_signing_alg_values_supported,
+            audience=proxied_authenticator.audience,
+            issuer=proxied_authenticator.issuer,
+        )
```

Comment:
```suggestion
    if proxied_authenticator:
        return jwt.decode(
            token,
            key=proxied_authenticator.keys,
            algorithms=proxied_authenticator.id_token_signing_alg_values_supported,
            audience=proxied_authenticator.audience,
            issuer=proxied_authenticator.issuer,
        )
```

nit: assigning to payload prior to the secret_keys loop is a change in behaviour if secret_keys contains only one element and it raises a JWTError, previously it would try and return payload before it is assigned and raise an exception, now it would return an empty string.